src/power_button.cpp

```diff
@@ -16,27 +16,313 @@
 
 #include "power_button.hpp"
 
+#include "utils.hpp"
+
+#include <fmt/format.h>
+
+#include <iomanip>
+#include <sstream>
+
 // add the button iface class to registry
 static ButtonIFRegister<PowerButton> buttonRegister;
 
-void PowerButton::simPress()
+using namespace phosphor::logging;
+
+void PowerButton::updatePressedTime()
 {
-    pressed();
+    pressedTime = std::chrono::steady_clock::now();
 }
 
-void PowerButton::simLongPress()
+void PowerButton::updatePressedTime(std::chrono::milliseconds timeout)
 {
-    pressedLong();
+    pressedTime = std::chrono::steady_clock::now() + timeout;
 }
 
-void PowerButton::updatePressedTime()
+auto PowerButton::getPressTime()
 {
-    pressedTime = std::chrono::steady_clock::now();
+    return pressedTime;
 }
 
-auto PowerButton::getPressTime() const
+std::string PowerButton::getPowerOpStateName(const PowerOpState& state)
 {
-    return pressedTime;
+    switch (state)
+    {
+        case PowerOpState::buttonNotPressed:
+            return "buttonNotPressed";
+        case PowerOpState::buttonPressed:
+            return "buttonPressed";
+        case PowerOpState::dpoInitiated:
+            return "dpoInitiated";
+        case PowerOpState::dpoFpoSeparation:
+            return "dpoFpoSeparation";
+        case PowerOpState::fpoInitiated:
+            return "fpoInitated";
+        default:
+            return "unknownState";
+    }
+}
+
+void PowerButton::buttonPressed()
+{
+    try
+    {
+        // Turn On the system if it is Off
+        if (!phosphor::button::utils::poweredOn())
+        {
+            log<level::DEBUG>(
+                fmt::format("{}", std::string(__func__) +
+                                      ": Power Button State - " +
+                                      getPowerOpStateName(powerOpState))
+                    .c_str());
+            released();
+            return;
+        }
+    }
+    catch (const sdbusplus::exception::exception& e)
+    {
+        log<level::DEBUG>(
+            fmt::format("{}: Exception - {}", std::string(__func__), e.what())
+                .c_str());
+        return;
+    }
+
+    std::stringstream ss;
+
+    // System is On, initiate the power Off process
+    if (powerOpState == PowerOpState::buttonNotPressed)
```

Comment:
I was thinking you'd do all of this in the button handler, as this class already emits signals on when buttons are pressed and released.

Reply:
Was thinking about that, but then I thought button handler is meant for handling the signal to power on/off purpose only. Will move this to button handler.